contrib/exporters/secadvisor/mod/encode_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"testing"
+
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/contrib/exporters/core"
+)
+
+func getEncoder(t *testing.T) core.Encoder {
+	cfg := viper.New()
+	encoder, err := NewEncodeSecadvsiorJSON(cfg)
+	if err != nil {
+		t.Fatalf("NewEncodeSecadvsiorJSON returned unexpected error: %v", err)
+	}
+	return encoder.(core.Encoder)
+}
+
+func Test_Encode_empty_flows_array(t *testing.T) {
+	in := make([]interface{}, 0)
+	result, err := getEncoder(t).Encode(in)
+	if err != nil {
+		t.Fatalf("Encode returned unexpected error: %v", err)
+	}
+	assertEqual(t, "{\"data\":[]}\n", string(result))
+}
+
+func Test_Encode_flows_array_with_objects(t *testing.T) {
+	in := []interface{}{
+		&SecurityAdvisorFlow{
+			Status: "STARTED",
+			Start: 1234,
+		},
+		&SecurityAdvisorFlow{
+			Status: "ENDED",
+			Last: 5678,
+		},
+	}
+	result, err := getEncoder(t).Encode(in)
+	if err != nil {
+		t.Fatalf("Encode returned unexpected error: %v", err)
+	}
+	assertEqual(t, `{"data":[{"Status":"STARTED","Start":1234,"Last":0,"UpdateCount":0},{"Status":"ENDED","Start":0,"Last":5678,"UpdateCount":0}]}` + "\n", string(result))
```

Comment:
you must check for JSON equality (not string equality) - for example if I change default for JSON encoder to pretty - I don't want to break this test.

For that end you can use something like:


```
func AreEqualJSON(s1, s2 string) (bool, error) {
	var o1 interface{}
	var o2 interface{}

	var err error
	err = json.Unmarshal([]byte(s1), &o1)
	if err != nil {
		return false, fmt.Errorf("Error mashalling string 1 :: %s", err.Error())
	}
	err = json.Unmarshal([]byte(s2), &o2)
	if err != nil {
		return false, fmt.Errorf("Error mashalling string 2 :: %s", err.Error())
	}

	return reflect.DeepEqual(o1, o2), nil
}
```

Reply:
format expected output string to:

```
expected := \
`{  
   "data":[  
      {  
         "Status":"STARTED",
         "Start":1234,
         "Last":0,
         "UpdateCount":0
      },
      {  
         "Status":"ENDED",
         "Start":0,
         "Last":5678,
         "UpdateCount":0
      }
   ]
}`
```